app/code/local/Mobbex/Mobbex/Block/Finance/Widget.php

```diff
@@ -3,48 +3,48 @@
 class Mobbex_Mobbex_Block_Finance_Widget extends Mage_Core_Block_Template
 {
 
+    public $styles = [];
 
     public function _construct()
     {
         parent::_construct();
+        $this->setTemplate('mobbex/finance.phtml');
 
-        $this->settings        = Mage::helper('mobbex/settings');
-        $this->mobbex          = $this->settings->helper;
-        $this->checkoutSession = Mage::getSingleton('checkout/session');
-        $this->action          = strpos(Mage::helper('core/url')->getCurrentUrl(), 'cart') ? 'cart' : 'product';
-
+        // Init class properties
+        \Mage::helper('mobbex/instantiator')->setProperties($this, ['sdk', 'settings', '_checkoutSession']);
+        
         //init widget
+        $this->action = strpos(Mage::helper('core/url')->getCurrentUrl(), 'cart') ? 'cart' : 'product';
         $this->initWidget();
+        foreach (['theme_type', 'button_text', 'button_logo', 'widget_style'] as $property)
```

Comment:
No será mucho hacer un foreach para esto? 😁 Creo que podríamos usar `$this->settings` directamente adentro del template, lo veo más sencillo